File: dotnet/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example1/Handlers/ShippingHandler.cs
namespace ChainOfResponsibilityDesignPattern.Example1.Handlers;

public class ShippingHandler : BaseOrderHandler
{
    private readonly Random _random = new();

    public override void Handle(OrderRequest request)
    {
        var shippingMethod = request.TotalAmount > 1000 ? "Express" : "Standard";
        var trackingNumber = $"TRK{_random.Next(100000, 999999)}";

        request.AddMessage($"Shipping arranged: {shippingMethod} ({trackingNumber})");
        request.IsApproved = true;
        
        base.Handle(request);
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example1/src/Handlers/ShippingHandler.ts
import { BaseOrderHandler } from "../BaseOrderHandler";
import { OrderRequest } from "../OrderRequest";

export class ShippingHandler extends BaseOrderHandler {
	handle(request: OrderRequest): void {
		const method = request.totalAmount > 1000 ? "Express" : "Standard";
		const tracking = `TRK${Math.floor(Math.random() * 900000) + 100000}`;
		request.addMessage(`Shipping arranged: ${method} (${tracking})`);
		request.isApproved = true;
		super.handle(request);
	}
}
